connection.js

```diff
@@ -1573,9 +1573,28 @@ Connection.prototype.queue_msg = function (retval, msg) {
     }
 };
 
+Connection.prototype.store_queue_result = function (retval, msg) {
+    var plugin = {name: 'queue'};
+    switch (retval) {
+        case constants.ok:
+            this.transaction.results.add(plugin, { pass: msg });
+            break;
+        case constants.deny:
+        case constants.denydisconnect:
+        case constants.denysoft:
+        case constants.denysoftdisconnect:
+            this.transaction.results.add(plugin, { fail: msg });
+            break;
+        default:
+            this.transaction.results.add(plugin, { msg: msg });
```

Comment:
That's not how Haraka works though.

hook_queue is only called when connection.relaying = false otherwise
hook_queue_outbound is called.

There must be a hook_queue plugin that calls OK otherwise the message is
rejected.
On 6 Jun 2015 11:14 pm, "celesteking" <notifications@github.com> wrote:

> In connection.js
> <https://github.com/baudehlo/Haraka/pull/950#discussion_r31872017>:
>
> > @@ -1573,9 +1573,28 @@ Connection.prototype.queue_msg = function (retval, msg) {
> >      }
> >  };
> >
> > +Connection.prototype.store_queue_result = function (retval, msg) {
> > +    var plugin = {name: 'queue'};
> > +    switch (retval) {
> > +        case constants.ok:
> > +            this.transaction.results.add(plugin, { pass: msg });
> > +            break;
> > +        case constants.deny:
> > +        case constants.denydisconnect:
> > +        case constants.denysoft:
> > +        case constants.denysoftdisconnect:
> > +            this.transaction.results.add(plugin, { fail: msg });
> > +            break;
> > +        default:
> > +            this.transaction.results.add(plugin, { msg: msg });
>
> With CONT returned (Haraka handles outbound), queue results object will
> be logged with msg set to null.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/baudehlo/Haraka/pull/950/files#r31872017>.
>